lib/rex/exploitation/cmdstager/bourne.rb

```diff
@@ -12,14 +12,18 @@ class CmdStagerBourne < CmdStagerBase
   def initialize(exe)
     super
 
-    @var_encoded = Rex::Text.rand_text_alpha(5)
-    @var_decoded = Rex::Text.rand_text_alpha(5)
+    @var_encoded = Rex::Text.rand_text_alpha(5) + '.b64'
+    @var_decoded = Rex::Text.rand_text_alpha(5) + '.bin'
   end
 
   def generate(opts = {})
     opts[:temp] = opts[:temp] || '/tmp/'
     opts[:temp] = opts[:temp].gsub(/'/, "\\\\'")
     opts[:temp] = opts[:temp].gsub(/ /, "\\ ")
+    if (opts[:file])
+        @var_encoded = opts[:file] + '.b64'
+        @var_decoded = opts[:file]
```

Comment:
Any reason you're not using ```.bin``` here, too? Alternatively, why are you using it above?